docs/src/devdocs/interpolations.md

```diff
@@ -22,30 +61,136 @@ Ferrite.reference_shape_hessians_gradients_and_values!
 Ferrite.shape_value_type(ip::Interpolation, ::Type{T}) where T<:Number
 ```
 
-### Required methods to implement for all subtypes of `Interpolation` to define a new finite element
+## [How to implement a new interpolation, `QTI`](@id devdocs-howto_new-interpolation)
+As an example, we will implement a `Q`uadratic `T`riangle `I`nterpolation
+(`QTI`), corresponding to `Lagrange{RefTriangle, 2}()`. This is described
+in the devdocs as the API is not considered fully stable yet.
 
-Depending on the dimension of the reference element the following functions have to be implemented
+```@example InterpolationExample
+using Ferrite # hide
+using Test #hide
+struct QTI <: ScalarInterpolation{RefTriangle, 2}
+end
+ip_qti = QTI() # hide
+ip_lag = Lagrange{RefTriangle, 2}() # hide
+nothing # hide
+```
+Before we jump in and define the shape functions associated with this
+interpolation, we first need to consider the `RefTriangle` entities,
 
-```@docs
-Ferrite.reference_shape_value(::Interpolation, ::Vec, ::Int)
-Ferrite.vertexdof_indices(::Interpolation)
-Ferrite.dirichlet_vertexdof_indices(::Interpolation)
-Ferrite.facedof_indices(::Interpolation)
-Ferrite.dirichlet_facedof_indices(::Interpolation)
-Ferrite.facedof_interior_indices(::Interpolation)
-Ferrite.edgedof_indices(::Interpolation)
-Ferrite.dirichlet_edgedof_indices(::Interpolation)
-Ferrite.edgedof_interior_indices(::Interpolation)
-Ferrite.volumedof_interior_indices(::Interpolation)
-Ferrite.getnbasefunctions(::Interpolation)
-Ferrite.reference_coordinates(::Interpolation)
-Ferrite.is_discontinuous(::Interpolation)
-Ferrite.adjust_dofs_during_distribution(::Interpolation)
-Ferrite.mapping_type
+```@docs; canonical=false
+RefTriangle
 ```
 
-for all entities which exist on that reference element. The dof functions default to having no
-dofs defined on a specific entity. Hence, not overloading of the dof functions will result in an
-element with zero dofs. Also, it should always be double checked that everything is consistent as
-specified in the docstring of the corresponding function, as inconsistent implementations can
-lead to bugs which are really difficult to track down.
+For this particular interpolation, we have one degree of freedom associated
+with each vertex, and one degree of freedom associated with each edge.
+Following the Ferrite numbering rules, we start by enumerating the
+vertices first, followed by the edges.
+```@example InterpolationExample
+function Ferrite.reference_shape_value(ip::QTI, ξ::Vec{2}, shape_number::Int)
+    ξ₁ = ξ[1]
+    ξ₂ = ξ[2]
+    γ = 1 - ξ₁ - ξ₂ # Helper
+    shape_number == 1 && return ξ₁ * (2ξ₁ - 1) # v1: 1 at ξ = (1, 0)
+    shape_number == 2 && return ξ₂ * (2ξ₂ - 1) # v2: 1 at ξ = (0, 1)
+    shape_number == 3 && return γ * (2γ - 1)   # v3: 1 at ξ = (0, 0)
+    shape_number == 4 && return 4ξ₁ * ξ₂ # e1: 1 at ξ = (½, ½)
+    shape_number == 5 && return 4ξ₂ * γ  # e2: 1 at ξ = (0, ½)
+    shape_number == 6 && return 4ξ₁ * γ  # e3: 1 at ξ = (½, 0)
```

Comment:
Maybe I have missed this, but do you reference somewhere where you got this specific numbering and coordinate information from (for the readers)?

Reply:
Not directly, could perhaps be more explicit. But the idea is that it comes from the reference shape that I show at the start (at
```Julia
```@docs; canonical=false
 RefTriangle
```

Reply:
I just want to make sure that we have some cross-references between the pages, so users will have an easier time to find the relevant information they need beyond the examples themselves. :)